inventory_management_system_api/services/item.py

```diff
@@ -118,9 +122,67 @@ def get(self, item_id: str) -> Optional[ItemOut]:
         :param item_id: The ID of the item to retrieve
         :return: The retrieved item, or `None` if not found
         """
-
         return self._item_repository.get(item_id)
 
+    def update(self, item_id: str, item: ItemPatchRequestSchema) -> ItemOut:
+        """
+        Update an item by its ID.
+
+        The method checks if the item exists in the database and raises a `MissingRecordError` if it does
+        not. If the system ID is being updated, it checks if the system ID with such ID exists and raises
+        a `MissingRecordError` if it does not. It raises a `ChildElementsExistError` if a catalogue item
+        ID is supplied. When updatimg properties, existing properties must all be supplied, or they will
+        be overwritten by the catalogue item properties.
+
+        :param item_id: The ID of the item to update.
+        :param item: The item containing the fields that need to be updated.
+        :return: The updated item.
+        """
+        update_data = item.model_dump(exclude_unset=True)
+
+        stored_item = self.get(item_id)
+        if not stored_item:
+            raise MissingRecordError(f"No item found with ID: {item_id}")
+
+        if "catalogue_item_id" in update_data and item.catalogue_item_id != stored_item.catalogue_item_id:
+            raise ChildElementsExistError("Cannot change the catalogue item the item belongs to")
```

Comment:
This exception is not relevant for this case. I discussed this with Joel and we think that it would be better if we create a new custom exception `InvalidActionError`, raise that here and handle it in the endpoint.

Reply:
what http status code should I use for it? still 422?

Reply:
I think so. We can change it later if we are proven wrong as it is usually the case with HTTP status codes 😅